build.gradle

```diff
@@ -8,14 +8,32 @@ plugins {
     id "java"
     id 'maven'
     id 'signing'
+    id 'jacoco'
     id 'com.palantir.git-version' version '0.5.1'
     id 'com.github.johnrengelman.shadow' version '1.2.3'
+    id "com.github.kt3k.coveralls" version "2.6.3"
 }
 
 repositories {
     mavenCentral()
 }
 
+jacocoTestReport {
+    dependsOn test
+    group = "Reporting"
+    description = "Generate Jacoco coverage reports after running tests."
+    additionalSourceDirs = files(sourceSets.main.allJava.srcDirs)
+
+    reports {
+        xml.enabled = true // coveralls plugin depends on xml format report
+        html.enabled = true
+    }
+}
+
+jacoco {
+    toolVersion = "0.7.5.201505241946"
+}
```

Comment:
I know very little about gradle itself, so I will trust that this works, or should we find someone else?